src/main/java/frc/robot/subsystems/LEDManager.java

```diff
@@ -26,34 +29,69 @@
  */
 public class LEDManager {
 
+    private static LEDManager instance;
+
     // Hardware
     private AddressableLED ledStrip;
     private AddressableLEDBuffer ledBuffer;
 
     /**
-     * Creates a new LEDManager object using the provided PWM port number.
+     * returns the singleton object of the LED Manager
+     * @return the singleton object
+     */
+    public static LEDManager getInstance() {
+        if (instance == null) {
+            throw new IllegalStateException("LEDManager has not been initialized. Call initialize() first.");
+        } else {
+            return instance;
+        }
+    }
+
+    /**
+     * initialized the LEDManager object using the provided PWM port number and length.
+     * 
+     * @param port The port number the LED strip is plugged into on the PWM.
+     * @param length The number of LEDs in the strip
+     */
+    public static void initialize(int port, int length) {
+        if (instance != null) {
+            throw new IllegalStateException("LEDManager has already been initialized.");
```

Comment:
In general, unless re-initializing would be destructive (or, in the method above, if initializing at time of use), I would recommend logging things as a warning, instead of throwing which will kill the program.